app/policies/sample_policy.rb

```diff
@@ -34,4 +34,37 @@ def destroy_attachment? # rubocop:disable Metrics/AbcSize
         )
       )
   end
+
+  scope_for :relation, :exportable_namespace_samples do |relation, options| # rubocop:disable Metrics/BlockLength
```

Comment:
**nitpick (blocking):** On further consideration I would like this to be refactored to `namespace_samples` and no longer pass in `sample_ids` but instead optionally pass in `minimum_access_level` and set that to `Member::AccessLevel::GUEST` by default.

That does mean you will need to call `Member.effective_access_level(namespace, user) >= minimum_access_level` instead of checking `can_export_data?`.

Also since you will no longer be passing in `sample_ids` you will need to do `authorized_scope(...).where(id: sample_ids)` in `app/services/data_exports/create_service.rb`.

Doing all this means that we will be able to remove `group_samples` scope and change all calls to that to use `namespace_samples`.